Pod/Classes/LSSpan.m

```diff
@@ -38,207 +35,170 @@ - (NSDictionary*) toJSON:(NSUInteger)maxPayloadJSONLength {
 
 @end
 
-@implementation LSSpan {
-    LSTracer* m_tracer;
-    LSSpanContext* m_ctx;
-    LSSpanContext* m_parent;
-    NSString* m_operationName;
-    NSDate* m_startTime;
-    NSMutableArray<LSLog*>* m_logs;
-    NSMutableDictionary* m_tags;
-}
-
-- (instancetype) initWithTracer:(LSTracer*)client {
-    return [self initWithTracer:client
-                  operationName:@""
-                         parent:nil
-                           tags:nil
-                      startTime:nil];
-}
-
-- (instancetype) initWithTracer:(LSTracer*)tracer
-                  operationName:(NSString*)operationName
-                         parent:(nullable LSSpanContext*)parent
-                           tags:(nullable NSDictionary*)tags
-                      startTime:(nullable NSDate*)startTime {
+#pragma mark - LSSpan
+
+@interface LSSpan ()
+@property(nonatomic, strong) LSSpanContext *parent;
+@property(atomic, strong) NSString *operationName;
+@property(atomic, strong) LSSpanContext *context;
+@property(nonatomic, strong) NSMutableArray<LSLog *> *logs;
+@property(atomic, strong, readonly) NSMutableDictionary<NSString *, NSString *> *mutableTags;
+@end
+
+@implementation LSSpan
+
+- (instancetype)initWithTracer:(LSTracer *)client {
+    return [self initWithTracer:client operationName:@"" parent:nil tags:nil startTime:nil];
+}
+
+- (instancetype)initWithTracer:(LSTracer *)tracer
+                 operationName:(NSString *)operationName
+                        parent:(nullable LSSpanContext *)parent
+                          tags:(nullable NSDictionary *)tags
+                     startTime:(nullable NSDate *)startTime {
     if (self = [super init]) {
-        self->m_tracer = tracer;
-        self->m_operationName = operationName;
-        self->m_startTime = startTime;
-        self->m_tags = [NSMutableDictionary dictionary];
-        self->m_logs = nil;
-
-        if (startTime == nil) {
-            m_startTime = [NSDate date];
-        }
-        UInt64 traceId = (parent == nil) ? [LSUtil generateGUID] : parent.traceId;
+        _tracer = tracer;
+        _operationName = operationName;
+        _startTime = startTime ?: [NSDate date];
+        _logs = @[].mutableCopy;
+        _mutableTags = @{}.mutableCopy;
+        _parent = parent;
+
+        UInt64 traceId = parent.traceId ?: [LSUtil generateGUID];
```

Comment:
@bensigelman Just FYI here:
- If a method returns a value number (non object), nil.foo = 0
- If a method returns a pointer (including NSNumber), nil.foo = nil
- Invoking anything on a `nil` is 100% safe and results in behavior above. In general, it returns a null-value
- Invoking anything on a `NSNull` crashes your application. Watch out on nulls received from APIs (in collections like dictionaries and arrays)